tests/test_single_topology_confgen.py

```diff
@@ -80,21 +79,40 @@ def run_edge(mol_a, mol_b, protein_path):
 
 
 @pytest.mark.nightly(reason="Takes a while to run")
-def test_confgen():
+def test_confgen_hard_edges():
+    # 1) cherry pick a couple of edges that are hard to setup initial geometries
+    # 2) failures come from failed edges in the original hif2a set
+    edges = [
+        ("30", "25"),  # core-hopping B-ring
+        ("227", "266"),  # core-hopping A-ring, bicycle
+        ("7a", "224"),  # core-hopping A-ring, 6->5 member,
+        ("227", "256"),  # stereo close-proximity down H with up H on B-ring,
+        ("289", "61"),  # stereo close-proximity down H with up H on B-ring,
+        ("266", "165"),  # core-hopping A-ring, and stereo close-proximity Hs on B-ring,
+        ("234", "227"),  # failure, stereo close proximity
+        ("1", "155"),  # failure, single nitrile mutation
+        ("1", "7a"),  # failure, methyl extension on B-ring
+        ("266", "156"),  # failure, A-ring corehopping unicycle -> bicycle, B ring double fluorination
+        ("289", "224"),  # failure, A-ring, 6->5 member
+        ("165", "42"),  # failure, B-ring, core-hopping,
+        ("7b", "42"),  # failure, B-ring, core-hopping,
+        ("35", "84"),  # failure, B-ring, core-hopping into oxazole,
+        ("252", "290"),  # failure, should be a simple transformation
+        ("290", "84"),  # failure, B-ring, core-hopping into oxazole
+        ("290", "256"),  # failure, should be a simple transformation
+        ("164", "163"),  # failure, B-ring has a different conformation
+    ]
 
     protein_path = "timemachine/testsystems/data/hif2a_nowater_min.pdb"
     ligands = "timemachine/datasets/fep_benchmark/hif2a/ligands.sdf"
     mols = [mol for mol in Chem.SDMolSupplier(ligands, removeHs=False)]
 
-    results_csv = "timemachine/datasets/fep_benchmark/hif2a/results_edges_5ns.csv"
-    with open(results_csv) as csvfile:
-        reader = csv.reader(csvfile, delimiter=",")
-        next(reader)
-        rows = [row for row in reader]
-        for row_idx, row in enumerate(rows):
-            mol_a_name, mol_b_name, exp_ddg, fep_ddg, fep_ddg_err, ccc_ddg, ccc_ddg_err = row
-            print("Processing", mol_a_name, "->", mol_b_name)
-            mol_a = get_mol_by_name(mols, mol_a_name)
-            mol_b = get_mol_by_name(mols, mol_b_name)
-
-            run_edge(mol_a, mol_b, protein_path)
+    n_windows = 16
+
+    for src, dst in edges:
+        print("\nProcessing", src, "->", dst, "\n")
+        mol_a = get_mol_by_name(mols, src)
+        mol_b = get_mol_by_name(mols, dst)
+        # try both directions
+        run_edge(mol_a, mol_b, protein_path, n_windows)
+        run_edge(mol_b, mol_a, protein_path, n_windows)
```

Comment:
Nit, testing -- still may be useful to add a unit test of `optimize_coordinates`, but this reduction of the nightly integration tests from 81 directed edges to 18*2=36 directed edges is good, and the annotations of why the selected edges were difficult is very helpful! 👍

Reply:
Also added a single unit test here as well.